client/pages/team/[id].vue

```
@@ -186,12 +130,98 @@
+                  class="border border-gray-200 rounded-lg p-4 bg-gray-50"
+                >
+                  <div class="flex items-start justify-between">
+                    <div class="flex-1">
+                      <h3 class="text-sm font-semibold text-gray-900">
+                        {{ assignment.rfcToBe?.draft.name }}
+                      </h3>
+                      <span>
+                        {{ assignment.comment || 'No Comment' }}
+                      </span>
+                      <div class="mt-2 flex items-center gap-4 text-xs text-gray-500">
+                        <span>
+                          Time spent: {{ assignment.timeSpent }}
+                        </span>
+                        <span v-if="assignment.state"
+                              :class="[
+                                assignment.state === 'done' ? 'text-green-600' :
+                                assignment.state === 'in_progress' ? 'text-blue-600' :
+                                'text-yellow-600'
+                              ]">
+                          {{ assignment.role }} ({{ assignment.state }})
+                        </span>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+
+          <!-- No assignments message -->
+          <div v-else class="mt-6 text-center py-8">
+            <p class="text-sm text-gray-500">No current assignments</p>
+          </div>
+        </div>
+        </div>
+
       </div>
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
+import type { RpcPerson } from '~/purple_client'
+
+const route = useRoute()
+const api = useApi()
+
+// Fetch person data
+const { data: person, pending, error } = await useAsyncData(
+  `person-${route.params.id}`,
+  async () => {
+    try {
+      const response = await api.rpcPersonRetrieve({ id: Number(route.params.id) })
+      return response
+    } catch (err) {
```

I wonder if we should try/catch or just throw so that the `error` is set.

---

@holloway just to understand, what is your concern with try/catch?

---

@rudimatz that it will hide any errors so the return value of ` { data: person, pending, error }` will never have `error` set for us to use in the Vue template. It's ok to let the API client throw exceptions in a `useAsyncData` afaik.

Alternatively when you catch it you could pass the `err` to `snackbarForErrors()`

---

ok fixed it